sun50i H6: Fix incorrect ram detection

viraniac opened the pull request:
# Description

Allwinner H6 boards like Orange Pi 3 LTS has this problem where u-boot spl sometimes detects the ram incorrectly as 4 GB instead of 2GB and because of that board doesn't boot any further until its forcefully power cylcled. I have noticed that adding a bit of delay in the ram detection function after calling mctl_core_init seems to fix the same. 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [X] Built and booted image on Orange Pi 3 LTS (sun50i H6). Changed default systemd target to reboot.target and left board rebooting overnight while serial console output was getting logged in a file. After I woke up, I noticed that board was still rebooting successfully without hanging and have performed 1350+ reboots.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

viraniac commented:
CC: @pyavitz @AGM1968

AGM1968 reviewed (approved):
Lgtm...

Matter of interest,  was there any discussion #linux sunxi?

Not important- curious

viraniac commented:
> Matter of interest, was there any discussion #linux sunxi?

No not really. I did the debugging and fixing locally. Previously I added the patch created by Megi from LibreElec source, but that was not working as expected. So I started debugging it myself, added few print statement where I have added the udelays and just adding print statement resolved the issue making me realize that we need to add some delays over there. 

I will try to upstream this and see what kind of feedback I receive

AGM1968 commented:
Excellent.... was going to suggest to consider upstreaming....

viraniac commented:
> Excellent.... was going to suggest to consider upstreaming....

My attempt to upstream this started a lot of discussion on U-boot mailing list for other potential fixes for this problem, while my patch was not upstreamed. But at the same time, I see that my changes are now used by LibreElec by Jernej
